app-api/src/main/java/org/openmrs/eip/app/CustomDatabaseOffsetBackingStore.java

```diff
@@ -0,0 +1,135 @@
+package org.openmrs.eip.app;
+
+import java.io.Serializable;
+import java.util.Date;
+import java.util.Map;
+
+import org.apache.commons.lang3.SerializationUtils;
+import org.apache.kafka.connect.runtime.WorkerConfig;
+import org.apache.kafka.connect.storage.Converter;
+import org.apache.kafka.connect.storage.MemoryOffsetBackingStore;
+import org.openmrs.eip.app.management.entity.DebeziumOffset;
+import org.openmrs.eip.app.management.repository.DebeziumOffsetRepository;
+import org.openmrs.eip.app.util.DebeziumOffsetUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import io.debezium.config.Instantiator;
+import io.debezium.embedded.EmbeddedEngine;
+
+@Component
+public class CustomDatabaseOffsetBackingStore extends MemoryOffsetBackingStore {
+	
+	private static final Logger log = LoggerFactory.getLogger(CustomDatabaseOffsetBackingStore.class);
+	
+	private static DebeziumOffsetRepository debeziumOffsetRepository;
+	
+	private String engineName;
+	
+	private Converter valueConverter;
+	
+	public CustomDatabaseOffsetBackingStore() {
+	}
+	
+	@Autowired
+	public CustomDatabaseOffsetBackingStore(DebeziumOffsetRepository debeziumOffsetRepository) {
+		CustomDatabaseOffsetBackingStore.debeziumOffsetRepository = debeziumOffsetRepository;
+	}
+	
+	@Override
+	public synchronized void start() {
+		super.start();
+		load();
+	}
+	
+	/**
+	 * @see {@link io.debezium.embedded.EmbeddedEngine#EmbeddedEngine()}
+	 */
+	@Override
+	public void configure(WorkerConfig config) {
+		Map<String, Object> originals = config.originals();
+		
+		this.engineName = (String) originals.get(EmbeddedEngine.ENGINE_NAME.name());
+		String valueConverterClassName = (String) originals.get(WorkerConfig.VALUE_CONVERTER_CLASS_CONFIG);
+		
+		Map<String, Object> valueConverterOriginalConfs = config
+		        .originalsWithPrefix(WorkerConfig.INTERNAL_VALUE_CONVERTER_CLASS_CONFIG + ".", true);
+		valueConverterOriginalConfs.put("schemas.enable", false);
+		
+		this.valueConverter = Instantiator.getInstance(valueConverterClassName, () -> this.getClass().getClassLoader(),
+		    null);
+		this.valueConverter.configure(valueConverterOriginalConfs, false);
+	}
+	
+	/**
+	 * @see
+	 */
+	private void load() {
+		long offsets = debeziumOffsetRepository.countByEnabledTrue();
+		
+		if (offsets > 0) {
+			log.info("Loading offset from the database");
+			DebeziumOffset offset = debeziumOffsetRepository.findFirstByEnabledTrueOrderByDateCreatedDesc();
+			
+			if (log.isDebugEnabled()) {
+				log.debug("Loaded offset: {}", offset);
+			}
+			
+			Map<byte[], byte[]> raw = (Map<byte[], byte[]>) SerializationUtils.deserialize(offset.getData());
```

Comment:
I actually thought you would capture the individual map entries and store them in a human readable format in the DB, but I guess it is not necessary